package.json

```diff
@@ -36,7 +36,7 @@
     "dev": "yarn workspace @apps/site dev",
     "dev:seed-db": "yarn workspace @apps/site dev:seed-db",
     "dev:db": "yarn workspace @apps/site dev:db",
-    "prepare": "patch-package --error-on-warn && turbo run build \"--filter=!./apps/*\""
+    "prepare": "patch-package --error-on-warn && turbo run codegen && turbo run build \"--filter=!./apps/*\""
```

Comment:
I'm trying to think of a better way around this and I can't think of one right now but this does feel like it's not aligning with how we should be using Turbo.

If the prepare step is dependent on codegen, isn't that something that should be represented in the Turbo dependency graph?

Also why do we need "site-data.json" if we're not building the site?

Reply:
> why do we need "site-data.json" if we're not building the site?

We don't `build` it but we do `lint:tsc` it, and it depends on the file to compile.

> If the prepare step is dependent on codegen, isn't that something that should be represented in the Turbo dependency graph?

Yes – I tried add this but it didn't work for some reason:
```json
    "lint:tsc": {
      "dependsOn": ["^build"],
      "outputs": []
    },
```
and because this is only restoring part of the `prepare` command that is in `main` anyway (which I removed in an earlier PR), I thought we could come back to it later, since it's an existing issue (merging this will mean no diff on this line from `0.3` to `main`)

Reply:
I'm fine with us revisiting later :)

Out of curiosity did you try:
```
    "lint:tsc": {
      "dependsOn": ["^codegen"],
      "outputs": []
    },
```

or 

```
    "lint:tsc": {
      "dependsOn": ["codegen"],
      "outputs": []
    },
```

I appreciate it's finicky to test

Reply:
This may be related to the fact that we exclude the frontend from the `turbo build` command. In any case it's extremely tedious and time consuming to test.

Reply:
I didn't try either of those things, no. We can do so as part of a general 'clean up dependency graphs in turbo' thing.

We can run `turbo something --graph` to get a visual representatoin to help debug